app/functoria_app.mli

```diff
@@ -88,8 +88,22 @@ module type S = sig
 
 end
 
-(** [Make] is an helper to generate new a application builder with the
-    custom constructs defined by {!S}. *)
+(** Configuration builder: stage 1 *)
+module ConfigureBuilder (P: S): sig
+  val run: unit -> unit
+  (** Run the configuration builder. This should be called exactly once
+      to run the configuration builder: command-line arguments will be
+      parsed, and some code will be generated and compiled. *)
+
+  val run_with_argv:
+    ?help_ppf:Format.formatter -> ?err_ppf:Format.formatter ->
+    string array -> unit
+  (** [run_with_argv a] is the same as {!run} but parses [a] instead
+      of the process command line arguments. It also allows to set
+      the error and help channels using [help_ppf] and [err_ppf]. *)
+end
```

Comment:
Is it really useful to create a new functor here? Do you see a use-case where users would just want to call `ConfigureBuilder` and where this would make sense to have it separate?

Reply:
The idea was to clearly separate the two stages of the build. So it's preferable to keep everything under the same module `Make` in order not to break API ?

Reply:
Yea I think it's preferable. I don't see how the new API is better for an user point of view (as we just have one user right now who is using both functors anyway)

Reply:
@samoht I updated the code accordingly